osu.Game/Users/UserListPanel.cs

```diff
@@ -67,6 +67,11 @@ protected override Drawable CreateLayout()
                             {
                                 username.Anchor = Anchor.CentreLeft;
                                 username.Origin = Anchor.CentreLeft;
+                            }),
+                            CreateGroupBadges().With(badges =>
```

Comment:
If you don't want fill flow container to add spacing when there are no badges then add the `GroupBadgeFlow` after the layout creation with `if (User.Groups?.Length > 0)`, similar to how supporter badge is added.